prov/gni/src/gnix_hashtable.c

```diff
@@ -715,9 +724,6 @@ int _gnix_ht_remove(gnix_hashtable_t *ht, gnix_ht_key_t key)
 
 void *_gnix_ht_lookup(gnix_hashtable_t *ht, gnix_ht_key_t key)
 {
-	if (ht->ht_state != GNIX_HT_STATE_READY)
-		return NULL;
-
```

Comment:
It's just a performance path issue. I'd rather make the assumption that we aren't doing something stupid and allow the lookup to occur without an additional branch.